Packages/com.unity.multiplayer.samples.coop/Utilities/Net/SessionManager.cs

```diff
@@ -232,36 +207,31 @@ public void OnSessionStarted()
         public void OnSessionEnded()
```

Comment:
OnSessionEnded is called in the OnNetworkSpawn for the post game state. What happens if we ever get a disconnect host side (I would guess some day we'll get that disconnect from relay if we lose our connection). Host would in theory brought back to main menu, and would skip the post game state. Which means the session would never get "ended" correct?

Reply:
and that point, why not have the session live inside the main scene as a gameobject? (it can still be accessible statically, but as soon as you start the session, sessionManager could spawn itself in the scene to follow the scene's lifecycle

Reply:
You're right, I did not think of that possibility.

I thought about your suggestions, and I think that the cleanest solution would be to simply make sure that Clear is called when the server ends (whether by requested disconnect or if disconnected by relay) and that this sets m_HasSessionStarted to false.

I still think it's better to not confine SessionManager to a gameobject so that it can have a lifetime longer than a scene. We need this to allow us to save player session data while they are connecting, and this makes it easier to reuse for projects where a session could last more than a single main scene.